datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -781,6 +781,16 @@ impl LogicalPlanBuilder {
         join_type: JoinType,
         is_all: bool,
     ) -> Result<LogicalPlan> {
+        let left_len = left_plan.schema().fields().len();
+        let right_len = right_plan.schema().fields().len();
+
+        if left_len != right_len {
+            return Err(DataFusionError::Plan(format!(
+                "Expected the no. of fields to be the same. Left fields count = {} and right fields count = {} are not the same.",
```

Comment:
```suggestion
                "each INTERSECT/EXCEPT query must have the same number of columns. Left fields count = {} and right fields count = {} are not the same.",
```

Reply:
I updated based on your suggestion. Let me know if that works.